app/Console/Commands/DailyChallengeQueueNext.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace App\Console\Commands;
+
+use App\Models\Multiplayer\DailyChallengeQueueItem;
+use App\Models\Multiplayer\Room;
+use App\Models\User;
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\DB;
+
+class DailyChallengeQueueNext extends Command
+{
+    protected $signature = 'daily-challenge:queue-next';
+
+    protected $description = 'Creates a new "daily challenge" multiplayer room for the first item in queue.';
+
+    public function handle()
+    {
+        $nextQueueItem = DailyChallengeQueueItem::whereNull('multiplayer_room_id')
+            ->orderBy('order', 'asc')
+            ->orderBy('id', 'asc')
+            ->first();
+
+        if ($nextQueueItem === null) {
+            $this->error('"Daily challenge" queue is empty.');
+            return 1;
+        }
+
+        $existing = Room::where('category', 'daily_challenge')->active()->first();
+        if ($existing !== null) {
+            $this->error("Another \"daily challenge\" room is open (id {$existing->id}).");
+            return 1;
+        }
+
+        DB::transaction(function () use ($nextQueueItem) {
+            $startTime = today();
+            $ownerId = $GLOBALS['cfg']['osu']['legacy']['bancho_bot_user_id'];
```

Comment:
I think it's more like `host` than `owner`

Reply:
Yeah there's "room host" but also "playlist item owner". Mixed these up a bit :sweat_smile: